File: file/rename.pl
#! /usr/bin/perl 

# rename.pl -- rename files by finding a fixed pattern and replace it by other
# Carlos Duarte, 031105

# a clone of the rename(1) utility present in most linux distributions

use strict; 

if (@ARGV < 3) {
	print "usage: $0 srcpat dstpat files...\n"; 
	exit 1; 
}

my $srcpat = shift;
my $dstpat = shift;

for (@ARGV) {
	my $old = $_;
	s/\Q$srcpat\E/$dstpat/; 
	my $new = $_;  
	$old eq $new and next; 
	print "$old -> $new\n"; 
	rename $old, $new; 
}

File: file/rename.py
#! /usr/bin/python
# rename.py -- rename files by finding a fixed pattern and replace it by other, python version
# Carlos Duarte, 060328

import sys, re, os

def main():
	if len(sys.argv) <= 3: 
		print "usage: %s srcpat dstpat files..." % sys.argv[0]
		sys.exit(1)
	srcpat = sys.argv[1]
	dstpat = sys.argv[2]
	pat = re.compile(re.escape(srcpat))
	for fn in sys.argv[3:]: 
		newfn,n = pat.subn(dstpat, fn)
		if newfn == fn: continue
		print fn, "->", newfn
		os.rename(fn, newfn)

if __name__ == "__main__": main()
